ccl/changefeedccl: unskip TestChangefeedOnlyInitialScanCSV for pulsar sink

ccl/changefeedccl.TestChangefeedOnlyInitialScanCSV [failed](https://tanzanite.cluster.engflow.com/invocations/default/e0d64c03-bbe8-4af3-b0cb-5866cef06f13?testReportRun=7&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [617bf347978dcc0d711399b1a76402d7f88de958](https://github.com/cockroachdb/cockroach/commits/617bf347978dcc0d711399b1a76402d7f88de958):


```
=== RUN   TestChangefeedOnlyInitialScanCSV
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedOnlyInitialScanCSV2394812205
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_test.go:7530: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedOnlyInitialScanCSV2394812205
--- FAIL: TestChangefeedOnlyInitialScanCSV (1.54s)
=== RUN   TestChangefeedOnlyInitialScanCSV/pulsar
    helpers_test.go:867: making server as system tenant
    helpers_test.go:943: making pulsar feed factory
    --- FAIL: TestChangefeedOnlyInitialScanCSV/pulsar (1.53s)
=== RUN   TestChangefeedOnlyInitialScanCSV/pulsar/initial_scan_only_with_csv
    changefeed_test.go:7513: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:7513
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:7525
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 2
        	Test:       	TestChangefeedOnlyInitialScanCSV/pulsar/initial_scan_only_with_csv
        --- FAIL: TestChangefeedOnlyInitialScanCSV/pulsar/initial_scan_only_with_csv (0.32s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScanCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36130

ccl/changefeedccl.TestChangefeedOnlyInitialScanCSV [failed](https://tanzanite.cluster.engflow.com/invocations/default/e71abb08-5df1-4731-8696-95790f0b9cf2?testReportRun=2&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb](https://github.com/cockroachdb/cockroach/commits/c9c3cc5f3c3a4a6ab556f4b9d5b6ec0381901bdb):


```
=== RUN   TestChangefeedOnlyInitialScanCSV
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedOnlyInitialScanCSV809831575
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_test.go:7530: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedOnlyInitialScanCSV809831575
--- FAIL: TestChangefeedOnlyInitialScanCSV (1.53s)
=== RUN   TestChangefeedOnlyInitialScanCSV/pulsar
    helpers_test.go:867: making server as system tenant
    helpers_test.go:943: making pulsar feed factory
    --- FAIL: TestChangefeedOnlyInitialScanCSV/pulsar (1.52s)
=== RUN   TestChangefeedOnlyInitialScanCSV/pulsar/initial_backfill_only_with_csv_multiple_tables
    changefeed_test.go:7513: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:7513
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:7525
        	Error:      	Not equal: 
        	            	expected: 6
        	            	actual  : 5
        	Test:       	TestChangefeedOnlyInitialScanCSV/pulsar/initial_backfill_only_with_csv_multiple_tables
        --- FAIL: TestChangefeedOnlyInitialScanCSV/pulsar/initial_backfill_only_with_csv_multiple_tables (0.30s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScanCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

ccl/changefeedccl.TestChangefeedOnlyInitialScanCSV [failed](https://tanzanite.cluster.engflow.com/invocations/default/78964d4c-58d5-4306-a532-70af90fa4fcb?testReportRun=18&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [7bb52a7d1c75d5adfdfa53e5fcff6f5e6497408f](https://github.com/cockroachdb/cockroach/commits/7bb52a7d1c75d5adfdfa53e5fcff6f5e6497408f):


```
=== RUN   TestChangefeedOnlyInitialScanCSV
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedOnlyInitialScanCSV311463542
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_test.go:7530: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedOnlyInitialScanCSV311463542
--- FAIL: TestChangefeedOnlyInitialScanCSV (1.63s)
=== RUN   TestChangefeedOnlyInitialScanCSV/pulsar
    helpers_test.go:867: making server as system tenant
    helpers_test.go:943: making pulsar feed factory
    --- FAIL: TestChangefeedOnlyInitialScanCSV/pulsar (1.62s)
=== RUN   TestChangefeedOnlyInitialScanCSV/pulsar/initial_backfill_only_with_csv
    changefeed_test.go:7513: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:7513
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:7525
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 2
        	Test:       	TestChangefeedOnlyInitialScanCSV/pulsar/initial_backfill_only_with_csv
        --- FAIL: TestChangefeedOnlyInitialScanCSV/pulsar/initial_backfill_only_with_csv (0.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=18</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOnlyInitialScanCSV.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -5941,6 +5941,9 @@ func TestChangefeedContinuousTelemetry(t *testing.T) {
 		for i := 0; i < 5; i++ {
 			beforeCreate := timeutil.Now()
 			sqlDB.Exec(t, fmt.Sprintf(`INSERT INTO foo VALUES (%d) RETURNING cluster_logical_timestamp()`, i))
+			// Read the event from the sink.
+			_, err := foo.Next()
+			require.NoError(t, err)
 			verifyLogsWithEmittedBytesAndMessages(t, jobID, beforeCreate.UnixNano(), interval.Nanoseconds(), false)
 		}
 	}
@@ -6007,12 +6010,20 @@ func TestChangefeedContinuousTelemetryDifferentJobs(t *testing.T) {
 		beforeInsert := timeutil.Now()
 		sqlDB.Exec(t, `INSERT INTO foo VALUES (1)`)
 		sqlDB.Exec(t, `INSERT INTO foo2 VALUES (1)`)
+		// Read the events from the sinks.
+		_, err := foo.Next()
+		require.NoError(t, err)
+		_, err = foo2.Next()
+		require.NoError(t, err)
 		verifyLogsWithEmittedBytesAndMessages(t, job1, beforeInsert.UnixNano(), interval.Nanoseconds(), false)
 		verifyLogsWithEmittedBytesAndMessages(t, job2, beforeInsert.UnixNano(), interval.Nanoseconds(), false)
 		require.NoError(t, foo.Close())
 
 		beforeInsert = timeutil.Now()
 		sqlDB.Exec(t, `INSERT INTO foo2 VALUES (2)`)
+		// Read the events from the sinks.
+		_, err = foo2.Next()
+		require.NoError(t, err)
 		verifyLogsWithEmittedBytesAndMessages(t, job2, beforeInsert.UnixNano(), interval.Nanoseconds(), false)
 		require.NoError(t, foo2.Close())
 	}
@@ -7538,8 +7549,7 @@ func TestChangefeedOnlyInitialScanCSV(t *testing.T) {
 		}
 	}
 
-	// TODO(#119289): re-enable pulsar
-	cdcTest(t, testFn, feedTestEnterpriseSinks, feedTestOmitSinks("pulsar"))
+	cdcTest(t, testFn, feedTestEnterpriseSinks)
 }
 
 func TestChangefeedOnlyInitialScanCSVSinkless(t *testing.T) {

--- a/pkg/ccl/changefeedccl/testfeed_test.go
+++ b/pkg/ccl/changefeedccl/testfeed_test.go
@@ -2584,7 +2584,8 @@ type mockPulsarServer struct {
 
 func makeMockPulsarServer() *mockPulsarServer {
 	return &mockPulsarServer{
-		msgCh: make(chan *pulsar.ProducerMessage, 2048),
+		// TODO(#118899): Make msgCh a buffered channel for async message sending.
+		msgCh: make(chan *pulsar.ProducerMessage),
 	}
 }
 
@@ -2638,12 +2639,14 @@ func (p *mockPulsarProducer) SendAsync(
 	m *pulsar.ProducerMessage,
 	f func(pulsar.MessageID, *pulsar.ProducerMessage, error),
 ) {
+	log.Infof(ctx, "SendAsync start %d", len(p.pulsarServer.msgCh))
 	select {
 	case <-ctx.Done():
 		f(nil, m, ctx.Err())
 	case p.pulsarServer.msgCh <- m:
 		f(nil, m, nil)
 	}
+	log.Infof(ctx, "SendAsync end")
 }
 
 func (p *mockPulsarProducer) LastSequenceID() int64 {

